### ISSUE TITLE
Question: how to stop JSCover from launching browser?

### ISSUE THREAD

**Issue by dnihar** [tntim96/JSCover-maven-plugin#14]
Hello,

Am trying to use JSCover with a different test runner, which is headless javascript tests (not phantom or jasmine). So I first instrument all the code and then run the tests as a different execution from pom.xml. But seems like JSCover automatically tries to run tests using whatever driver config etc..which then throws exception.

[ERROR] Failed to execute goal com.github.tntim96:jscover-file-maven-plugin:1.0.15:jscover (default) on project aura-framework: Execution default of goal com.github.tntim96:jscover-file-maven-plugin:1.0.15:jscover failed: java.lang.reflect.InvocationTargetException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.

So all I want to know is to stop JSCover once it's done instrumenting.

**Comment by dnihar** [comment]
And this is using,
<groupId>com.github.tntim96</groupId>
<artifactId>jscover-file-maven-plugin</artifactId>
<version>1.0.15</version>

**Comment by tntim96** [comment]
The plugin tries to run the tests. I would need to write a new goal to just instrument the JavaScript. So you just want the JS instrumented as part of the `process-test-classes` phase?

**Comment by dnihar** [comment]
Exactly.

Also, I want to know, how can I exclude the test-scripts themselves from being instrumented? I added those to <excludeArgs> but then because JSCover tried to run tests, it broke since it could not find the tests there (as expected, because of exclude). The above goal would probably help since now, it won't kick off tests.

**Comment by tntim96** [comment]
It would be done using the `excludeArgs` as shown in the [REDME.md](https://github.com/tntim96/JSCover-maven-plugin/blob/master/plugin-parent/file-system/README.md).

**Comment by dnihar** [comment]
Yep, that's what I did, but then it said, 
Problem finding test pages: basedir /Users/myself/project/instrumented/project-impl/src/test ..because when I excluded, it did not even copy over ${project.home}/project-impl/src/test dir to under <reportDir>

                       <configuration>
                            <srcDir>${project.home}/project-impl</srcDir>
                            <testDirectory>${project.home}/project-impl/src/test</testDirectory>
                            <reportDir>${project.home}/instrumented/project-impl</reportDir>
                            <testIncludes>xunit-project-impl*.html</testIncludes>
                            <branchCoverageMinimum>100</branchCoverageMinimum>
                            <functionCoverageMinimum>100</functionCoverageMinimum>
                            <lineCoverageMinimum>100</lineCoverageMinimum>
                            <excludeArgs>
                                <excludeArg>--exclude=src/main/java</excludeArg>
                                <excludeArg>--exclude=src/test</excludeArg>
                                <excludeArg>--exclude=target</excludeArg>
                            </excludeArgs>

**Comment by tntim96** [comment]
Sorry, I gave you the wrong config. From the same example use the `--no-instrument` argument in `instrumentPathArgs`:
```xml
            <instrumentPathArgs>
                <arg>--no-instrument=main/webapp/js/vendor/</arg>
                <arg>--no-instrument=test</arg>
            </instrumentPathArgs>
```

**Comment by dnihar** [comment]
ok, let me try that.. I did see this as well, but then thought it would only resolve url endpoints as opposed to filesystem paths.

**Comment by tntim96** [comment]
The file-system paths are relative to the `srcDir`

**Comment by dnihar** [comment]
Ok, that worked, not exactly as I wanted, but works for now. Inside src/test I have 4 directories, 'lint', 'java', 'javascript', 'results' so I was doing a combination of --excludeArg to exclude 'lint' 'java' 'results' so that they don't even get copied over unnecessarily.. and then --no-instrument for src/test/javascript which has test scripts. But because of --excludeArg the entire src/test directory did not get generated. So for now, it's ok, using just --no-instrument works, replicating the entire structure under src/test

Thanks for your quick response, will wait on the goal we talked above, to just instrument resources.

**Comment by tntim96** [comment]
Want to try `1.0.16-SNAPSHOT` (with `jscover-file:instrument`) using Sonar plugin snapshot repository?

```xml
    <pluginRepositories>
        <pluginRepository>
            <id>central</id>
            <url>http://repo1.maven.org/maven2/</url>
        </pluginRepository>
        <!-- Testing snapshot development -->
        <pluginRepository>
            <id>Sonatype repository</id>
            <name>Sonatype's Maven repository</name>
            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
        </pluginRepository>
    </pluginRepositories>
```

**Comment by dnihar** [comment]
Thanks, that worked. I would have requested to make testDirectory optional, since if we are just going to instrument then that testDirectory should not matter in that configuration, but that's ok.

There is another issue I guess for which I will open up a new one. Good to close this one.
